CHANGES.md

```diff
@@ -1,3 +1,104 @@
+# 0.9.2
+
+This minor release contains various bug fixes, improvements, and a **breaking**
+change.
+
+## Breaking change: Trailing semicolons are no longer permitted
+The surface syntax of Links has been changed.  Up until now it was possible to
+end a block with a semicolon.  A trailing semicolon was interpreted as
+implicitly ending the block with a `()` expression.  The rationale for this
+change is to make the Links syntax more consistent, i.e. now all blocks must end
+with an explicit expression.  To sum up, previously both of the following were
+allowed
+
+```links
+fun foo(x) {
+  bar(y);
+  baz(x);
+}
+fun foo'(x) {
+  bar(y);
+  baz(x)
+}
+```
+
+Now the first form is no longer accepted. Instead you have to drop the semicolon
+and either end the block with an explicit `()` or wrap the last expression in an
+`ignore` application.
+
+```links
+fun foo(x) {
+  bar(y);
+  baz(x);
+  ()
+}
+
+fun foo(x) {
+  bar(y);
+  ignore(baz(x))
+}
+```
+
+A third option is to simply drop the trailing semicolon, though, this only works
+as intended if the type of the last expression is `()`.
+
+
+## SML-style function definitions
+
+Links now supports "switch functions", a new syntax for defining functions in
+terms of match clauses directly, similar to SML. This allows writing the
+following function
+```links
+fun ack(_,_) switch {
+  case (0, n) -> n + 1
+  case (m, 0) -> ack(m - 1, 1)
+  case (m, n) -> ack(m - 1, ack(m, n - 1))
+}
+```
+instead of the following, more verbose version:
+
+```links
+fun ack(a, b) {
+  switch(a, b) {
+    case (0, n) -> n + 1
+    case (m, 0) -> ack(m - 1, 1)
+    case (m, n) -> ack(m - 1, ack(m, n - 1))
+  }
+}
+```
+
+Switch functions can also be anonymous, allowing function like the following:
+```links
+fun(_, _) switch {
+  case (0, n) -> 0
+  case (m, n) -> m + n
+}
+```
+
+
```

Comment:
```suggestion
As switch functions have experimental status they are disabled by default. To enable them you must set the option `pattern_matching_sugar=true` in a configuration file. 
```

Reply:
I wonder whether we can make a swift change. Rather than naming the option "pattern_mathcing_sugar" it ought to be named "switch_funs" or something along those lines. 

A further thought would be to adopt an extension scheme similar to that of GHC, where an extension is toggled by passing `-Xextension_name` on the commandline. In fact, I deliberately engineered the settings framework to support this kind of interface. But let's leave this for a later discussion.

Reply:
Sure. Do you want to make the change yourself?

Reply:
I've opened a PR to rename the option "pattern_matching_sugar" to "switch_functions", c.f. #919